angular-client/src/pages/charging-page/components/fault-display/fault-display.component.ts

```diff
@@ -19,33 +20,29 @@ export default class FaultDisplay {
 
   ngOnInit() {
     this.storage.get(IdentifierDataType.COMM_TIMEOUT_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Communication Timeout', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.HARDWARE_FAILURE_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Hardware Failure', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.OVER_TEMP_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Over Temp', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.VOLTAGE_WRONG_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Voltage Wrong', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.WRONG_BAT_CONNECT_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Wrong Battery Connect', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
```

Comment:
you can just do parseInt here instead, this topic should only return integer values (same for every line). Also if you're going to pass the value into a helper method anyways, I'd also do the integer conversion in there